src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const grouped: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const itemKey = String(item[key]);
+
+    if (grouped[itemKey]) {
+      grouped[itemKey].push(item);
+    } else {
+      grouped[itemKey] = [item];
+    }
+  });
+
+  return grouped;
```

Comment:
suggest to use reduce method instead forEach.
and immediately return result of reduce.